### PR TITLE
Move IP related tests to ip.rs

### PR DESCRIPTION
**Pull request by achanda**
- Also move common functions to test.rs
- Leaves out Socket address related tests in addr.rs

### CODE REVIEW

File: src/libstd/net/test.rs [hunk 1]
```diff
@@ -27,6 +27,21 @@ pub fn next_test_ip6() -> SocketAddr {
...
+
+pub fn sa6(a: Ipv6Addr, p: u16) -> SocketAddr {
+        SocketAddr::V6(SocketAddrV6::new(a, p, 0, 0))
+}
+
+pub fn tsa<A: ToSocketAddrs>(a: A) -> Result<Vec<SocketAddr>, String> {
+        match a.to_socket_addrs() {
+            Ok(a) => Ok(a.collect()),
+            Err(e) => Err(e.to_string()),
+        }
```
Comment (alexcrichton) [inline src/libstd/net/test.rs]: It looks like these functions got an extra level of indentation perhaps?

**Comment by alexcrichton** [comment]
Awesome, thanks!

**Comment by achanda** [comment]
@alexcrichton fixed now

**Pull request merged**